Library/Homebrew/cmd/update.sh

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+
+if [ -z "$HOMEBREW_BREW_FILE" ]
+then
+  echo "Error: $(basename "$0") must be called from brew!" >&2
+  exit 1
+fi
+
+if [ "$(id -u)" = "0" ] && [ "$(stat -f%u "$HOMEBREW_BREW_FILE")" != "0" ]
+then
+  cat >&2 <<EOS
+Cowardly refusing to `sudo brew update`
+You can use brew with sudo, but only if the brew executable is owned by root.
+However, this is both not recommended and completely unsupported so do so at
+your own risk.
+EOS
+  exit 1
+fi
```

Comment:
Yeh, I guess I didn't want to handle the "check if a file is present in this array" but I guess that's not too much pain to do.